open-craft/opencraft
# Allows the OpenStack base image, flavor, and ssh_keyname to be configured per instance with django_extensions.db.fields.json.JSONField

This change reinstates the changes from https://github.com/open-craft/opencraft/pull/169, which were reverted by https://github.com/open-craft/opencraft/pull/174.

To fix the issues with stage/prd migrations, this change uses `django_extensions.db.fields.json.JSONField` instead of `django.contrib.postgres.fields.JSONField`, which requires PostgreSQL ≥ 9.4.

**Testing instructions**:

1. Checkout this PR's branch: `jill/configure_base_image-json-fix`
1. If you're running a devstack that includes the migrations from https://github.com/open-craft/opencraft/pull/169, you'll need to revert them.

    ```
    honcho run python3 manage.py migrate --fake instance 0072_openstack_settings
    honcho run python3 manage.py migrate instance 0071_auto_20170120_0826
    ```
1. Verify the change using the instructions on https://github.com/open-craft/opencraft/pull/169.

**Reviewers**
